src/promptflow/promptflow/executor/_base.py

```diff
@@ -0,0 +1,70 @@
+import abc
+from pathlib import Path
+from typing import Any, Callable, Dict, Mapping, Optional
+
+from promptflow._constants import LINE_TIMEOUT_SEC
+from promptflow.executor._flow_nodes_scheduler import DEFAULT_CONCURRENCY_FLOW
+from promptflow.executor._result import AggregationResult, LineResult
+from promptflow.storage import AbstractRunStorage
+
+
+class FlowExecutorBase(abc.ABC):
```

Comment:
why this base class.

FlowExecutor is just a one instance method inside one process. There will be no variant of FlowExecutor in python context.

Reply:
Maybe RunExecutor vs TestExecutor :D

Reply:
this is for further multi-language support, each language will have its own executor wrapper. It has only one subclass, FlowExecutor for python, for now